roadmap/roadmap_2024.md

```diff
@@ -0,0 +1,146 @@
+- Reduce the maintainability effort to generate the images used by Kubebuilder projects and dependency within third-party solutions.
+  - Related issues:
+    - [Issue #1885 - use a NetworkPolicy instead of kube-rbac-proxy](https://github.com/kubernetes-sigs/kubebuilder/issues/1885)
+    - [Issue #3230 - Migrate away from google.com gcp project kubebuilder](https://github.com/kubernetes-sigs/kubebuilder/issues/3230)
+
+**Proposed Solutions:**
+
+- **Making kube-rbac-proxy Optional:** Offering users the option to include kube-rbac-proxy caters to diverse project
+  requirements and simplifies the transition towards its potential externalization or removal,
+  reducing future maintenance efforts.
+
+- **Leveraging NetworkPolicies:** This alternative focuses on minimizing external dependencies by
+  utilizing Kubernetes-native solutions like NetworkPolicies, in line with our maintenance reduction goals.
+
+- **Default Enablement of cert-manager:** While not directly addressing the maintenance concerns related to
+  kube-rbac-proxy, defaulting to cert-manager responds to community feedback and navigates the upcoming deprecations.
+  This strategy also acknowledges cert-manager's existing role as a prerequisite for webhooks.
+
+---
+### Providing Helpers for Project Distribution
+
+#### Distribution via Kustomize
+
+**Status:** :white_check_mark: Complete
+
+As of release ([v3.14.0](https://github.com/kubernetes-sigs/kubebuilder/releases/tag/v3.14.0)),
+Kubebuilder includes enhanced support for project distribution.
+Users can now scaffold projects with a `build-installer` makefile target.
+This improvement enables the straightforward deployment of solutions directly to Kubernetes clusters.
+Users can deploy their projects using commands like:
+
+```shell
+kubectl apply -f https://raw.githubusercontent.com/<org>/my-project/<tag or branch>/dist/install.yaml
+```
+
+This enhancement streamlines the process of getting Kubebuilder projects running on clusters, providing a seamless deployment experience.
+
+#### (New Optional Plugin) Helm Chart Packaging
+
+**Status:** :raised_hands: Proposal in Progress; Seeking Contributions
+
+**Objective:** We aim to introduce a new plugin for Kubebuilder that packages projects as Helm charts,
+facilitating easier distribution and integration of solutions within the Kubernetes ecosystem. For details on this proposal and how to contribute,
+see [GitHub Pull Request #3632](https://github.com/kubernetes-sigs/kubebuilder/pull/3632).
+
+**Motivation:** The growth of the Kubernetes ecosystem underscores the need for flexible and
+accessible distribution methods. A Helm chart packaging plugin would simplify the distribution of the solutions
+and allow easily integrations with common applications used by administrators.
+
+---
+### Updating Scaffolding to Align with the Latest changes of controller-runtime
```

Comment:
We could also add that this involves updating dependencies based on k8s bumps. It would be nice to have community's help on this too.

Reply:
I have no objetions about we add a follow up like "Keep the project updated with the edge" so that we clarify that is a goal as well for all deps.

However, I think we have this goal already by default right. It might be implicit in keep the project maintained.